src/DGtal/io/writers/MagickWriter.h

```diff
@@ -0,0 +1,129 @@
+#define MagickWriter_RECURSES
+
+#if !defined MagickWriter_h
+/** Prevents repeated inclusion of headers. */
+#define MagickWriter_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <string>
+#include "DGtal/base/CUnaryFunctor.h"
+#include "DGtal/base/BasicFunctors.h"
+#include "DGtal/io/writers/MagickWriter.h"
+#include "DGtal/io/Color.h"
+
+#ifndef WITH_MAGICK
+#pragma error "You must activate imagemagick (-DWITH_MAGICK=true) to include this file"
+#endif
+#include <Magick++.h>
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // template class MagickWriter
+  /**
+   * Description of template struct 'MagickWriter' <p>
+   * \brief Aim: Export image using Imagemagick backend.
+   *
+   * If imagemagick is enabled (with WITH_MAGICK set to true), this writer
+   * export any image to a generic color Magick++ image (png, jpg, gif, bmp...) using
+   * a given colormap functor.
+   *
+   * The file format is given by the filename extension during the export.
+   *
+   * @tparam TImage the Image type.
+   * @tparam TFunctor the type of functor used in the export. The return type
+   * of the functor must be a DGtal::Color (e.g. any colormap)
+   *
+   * @see testMagickWriter.cpp
+   */
+  template <typename TImage, typename TFunctor =  functors::Identity>
+  struct MagickWriter
+  {
+    // ----------------------- Standard services ------------------------------
+    typedef TImage Image;
+    typedef typename TImage::Value Value;
+    typedef TFunctor Functor;
+    
+    BOOST_CONCEPT_ASSERT((  concepts::CUnaryFunctor<TFunctor, Value, Color> )) ;
+    BOOST_STATIC_ASSERT( (TImage::Domain::dimension == 2) );
+
+    /** 
+     * Export an Image with Imagemagick.
+     * 
+     * @param filename name of the output file the suffix is used to select
+     * the file format (png, bmp, jpg,...).
+     * @param anImage the image to export
+     * @param aFunctor  functor used to cast image values to DGtal::Color
+     *
+     * @return true if no errors occur.
+     */
+    static bool exportMagick(const std::string & filename, const Image &anImage,
+                             const Functor & aFunctor = Functor())
+    {
+      Magick::InitializeMagick(NULL);
+      int w = (anImage.domain().upperBound()[0] - anImage.domain().lowerBound()[0])+1;
+      int h = (anImage.domain().upperBound()[1] - anImage.domain().lowerBound()[1])+1;
+      Magick::Geometry geom(w,h);
+      Magick::Image image(geom,"white");
+      for(auto point: anImage.domain())
+      {
+        Color c = aFunctor( anImage(point) );
+        Magick::ColorRGB  magickc( c.red()/255., c.green()/255., c.blue()/255.) ;
+        image.pixelColor( point[0] + anImage.domain().lowerBound()[0],
+                         (h-1) - (point[1] + anImage.domain().lowerBound()[1]),
```

Comment:
anImage.domain().upperBound()[1]-point[1] ?

Reply:
Thanks , I’ll check

Reply:
that should be the same.